src/stores/mainStore.ts

```diff
@@ -117,6 +108,85 @@ export class MainStore {
     }
   }
 
+  handleInitialCardsStage() {
+    this.baccarat.givePlayerACard();
+    this.baccarat.givePlayerACard();
+    this.baccarat.giveBankerACard();
+    this.baccarat.giveBankerACard();
+
+    this.baccarat.flipPlayerCards();
+    this.baccarat.flipBankerCards();
+
+    const timer = setTimeout(() => {
+      this.game.setGameStage(GameStage.CheckForThirdCard);
+      clearTimeout(timer);
+    }, 2000);
+  }
+
+  handleGameRoundEnd() {
+    const winnerTimer = setTimeout(() => {
+      this.setWinner();
+      clearTimeout(winnerTimer);
+    }, 2000);
+    const timer = setTimeout(() => {
+      if (this.player.playerMoney < 1 || !this.baccarat.nextRoundIsPossible) {
+        this.endGameReset();
+      } else {
+        this.betweenRoundsReset();
+      }
+      this.game.setGameStage(GameStage.InitialBet);
+      clearTimeout(timer);
+    }, 4000);
+  }
+
+  handleThirdCard(receiver: "player" | "banker") {
```

Comment:
You have a lot of repetitive code with `setTimeout`. What you can do.

Extract setTimeout to a separate function
```typescript
function delay(fn: () => void, time: number) {
  const timer = seTimeout(() => {
    fn(),
    clearTimeout(timer);
  }, time)
}
```

And later use it like
```typescript
this.baccarat.givePlayerACard();
delay(() => {
  if (this.baccarat.playerCards[2]) {
    this.baccarat.flipThirdPlayerCard();
  }
}, 500)
```